hearth/lib/hearth/http/client.rb

```diff
@@ -36,15 +36,24 @@ class Client
       #   authority files for verifying peer certificates.  If you do
       #   not pass `:ssl_ca_bundle` or `:ssl_ca_directory` the
       #   system default will be used if available.
+      #
+      # @option options [OpenSSL::X509::Store] :ssl_ca_store An OpenSSL X509
+      #   certificate store that contains the SSL certificate authority.
+      #
+      # @option options [#resolve_address] (Hearth::DNS::HostResolver)
+      #   :dns_resolver An object that responds to `#resolve_address`, returning
+      #   an array of up to two IP addresses for the given hostname, one IPv6
+      #   and one IPv4, in that order. `#resolve_address` takes keyword args
+      #   similar to Addrinfo.getaddrinfo's positional parameters.
       def initialize(options = {})
         @http_wire_trace = options[:http_wire_trace]
         @logger = options[:logger]
-        @http_proxy = options[:http_proxy]
-        @http_proxy = URI.parse(@http_proxy.to_s) if @http_proxy
+        @http_proxy = URI(options[:http_proxy]) if options[:http_proxy]
         @ssl_verify_peer = options[:ssl_verify_peer]
         @ssl_ca_bundle = options[:ssl_ca_bundle]
         @ssl_ca_directory = options[:ssl_ca_directory]
         @ssl_ca_store = options[:ssl_ca_store]
+        @dns_resolver = options[:dns_resolver] || Hearth::DNS::HostResolver.new
```

Comment:
What about defaulting to nil and not replacing the default?  We can still allow it to be overwritten and can provide the interface and an implementation, but not change it by default?

Reply:
That might be a good option..